python/pyspark/sql/dataframe.py

```diff
@@ -1289,6 +1289,17 @@ def cast(self, dataType):
             raise TypeError("unexpected type: %s" % type(dataType))
         return Column(jc)
 
+    @ignore_unicode_prefix
+    def between(self, lowerBound, upperBound):
+        """ A boolean expression that is evaluated to true if the value of this
+        expression is between the given columns.
+
+        >>> df[df.col1.between(lowerBound, upperBound)].collect()
+        [Row(col1=5, col2=6, col3=8)]
+        """
+        jc = (self >= lowerBound) & (self <= upperBound)
+        return Column(jc)
```

Comment:
actually I think you no longer need to wrap it in Column, since it is already a Python column.